pallets/subspace/src/subnet.rs

```diff
@@ -523,4 +524,19 @@ impl<T: Config> Pallet<T> {
             .map(|(netuid, _)| netuid)
             .collect()
     }
+
+    pub fn remove_subnet_dangling_keys(netuid: u16) {
+        'outer: for (_, key) in Keys::<T>::iter_prefix(netuid) {
+            for (netuid_i, _, key_i) in Keys::<T>::iter() {
+                if netuid == netuid_i {
+                    continue;
+                }
+                if key == key_i {
+                    continue 'outer;
+                }
+            }
+
+            Self::remove_stake_from_storage(&key);
+        }
+    }
```

Comment:
You can use a BTreeSet and calculate the difference between the sets:
```suggestion
    pub fn remove_subnet_dangling_keys(netuid: u16) {
        let netuid_keys: BTreeSet<AccountId> = Uids::<T>::iter_prefix(netuid).collect();
        let global_keys: BTreeSet<AccountId> = Uids::<T>::iter().filter(|n, _, _| n != netuid).map(|_, key, _| key).collect();
        for dangling in netuid_keys.difference(global_keys) {
            Self::remove_stake_from_storage(&dangling);
        }
    }
```